orbit/pkg/user/user_darwin.go

```diff
@@ -0,0 +1,30 @@
+//go:build darwin
+// +build darwin
+
+package user
+
+import (
+	"errors"
+	"os/exec"
+	"strings"
+)
+
+// IsUserLoggedInViaGui returns whether or not a user is logged into the machine via the GUI.
+func IsUserLoggedInViaGui() (bool, error) {
+	output, err := exec.Command("/usr/bin/stat", "-f", "%Su", "/dev/console").Output()
+	if err != nil {
+		var ee *exec.ExitError
+		if errors.As(err, &ee) {
+			return false, errors.Join(err, errors.New(string(ee.Stderr)))
```

Comment:
Nice! Just in case there's nothing there though, I'd add a condition `errors.As(err, &ee) && len(ee.Stderr) > 0`.